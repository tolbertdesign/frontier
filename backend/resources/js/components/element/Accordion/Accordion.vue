<template>
  <div>
    <BCollapse
      :open="isOpen"
      class="card shadow-none border-t-2 border-grey-light"
    >
      <button
        slot="trigger"
        slot-scope="props"
        class="flex items-baseline w-full justify-between mb-px py-4 sm:py-4 px-4 md:px-4 cursor-pointer shadow-none focus:outline-none"
        @click="collapseClick"
      >
        <slot name="title">
          <p class="text-xl sm:text-2xl font-semibold">
            <span class="pl-1">{{ title }}</span>
          </p>
        </slot>
        <slot name="toggle">
          <span class="flex cursor-pointer items-center justify-center text-grey-dark">
            <FontAwesomeIcon :icon="props.open ? 'chevron-up' : 'chevron-down'" />
          </span>
        </slot>
      </button>
      <slot>
        <div class="p-2">
          <div class="modal-card-body bg-white p-4 md:p-6 text-sm sm:text-base md:text-lg">
            <p class="mb-4">
              {{ content }}
            </p>
          </div>
        </div>
      </slot>
    </BCollapse>
  </div>
</template>

<script>
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
export default {
  name: 'Accordion',
  status: 'prototype',
  release: '1.0.0',
  components: {
    FontAwesomeIcon,
  },
  props: {
    title: {
      type: String,
      default: '',
    },
    content: {
      type: String,
      default: '',
    },
    isOpen: {
      type: Boolean,
      default: false,
    },
    shouldReslick: {
      type: Boolean,
      default: false,
    },
  },
  methods: {
    collapseClick () {
      if (this.shouldReslick) {
        this.$nextTick(
          () => {
            if (navigator.userAgent.indexOf('MSIE') !== -1 || navigator.appVersion.indexOf('Trident/') > 0) {
              var evt = document.createEvent('UIEvents')
              evt.initUIEvent('reslicked', true, false, window, 0)
              window.dispatchEvent(evt)
            } else {
              window.dispatchEvent(new Event('reslicked'))
            }
          },
        )
      }
    },
  },
}
</script>
