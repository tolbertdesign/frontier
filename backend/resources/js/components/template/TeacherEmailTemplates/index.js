import TeacherEmailTemplates from './TeacherEmailTemplates.vue'

export default TeacherEmailTemplates
