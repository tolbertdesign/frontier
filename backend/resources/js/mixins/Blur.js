export default {
  methods: {
    blur () {
      document.getElementById('app').style.filter = 'blur(4px)'
    },
    unBlur () {
      document.getElementById('app').style.filter = 'none'
    },
  },
}
