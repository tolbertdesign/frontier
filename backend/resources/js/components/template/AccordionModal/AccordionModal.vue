<template>
  <div class="max-h-screen overflow-y-auto shadow-lg">
    <div class="modal-card rounded-xl bg-white shadow-lg">
      <header class="modal-card-head bg-white relative items-start">
        <slot name="header">
          <p class="modal-card-title flex items-center justify-between text-xl font-weight-normal">
            <span style="width: 24px; height: 48px" />
            <span class="text-2xl flex-1 text-center font-bold capitalize">{{ header }}</span>
            <a
              href="#"
              class="text-black self-start"
              @click.prevent="closeModal"
            >
              <span class="icon">
                <FontAwesomeIcon
                  :icon="['fas', 'times']"
                />
              </span>
            </a>
          </p>
        </slot>
      </header>
      <section class="modal-card-body p-0 overflow-auto">
        <slot>
          <div class="modal-card-body bg-white p-4 md:p-6 text-sm sm:text-base md:text-lg">
            <p class="mb-4">
              {{ content }}
            </p>
          </div>
        </slot>
      </section>
      <div class="card-footer">
        <div class="card-footer-item flex-col">
          <slot name="footer">
            {{ footer }}
          </slot>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import Blur from '@/mixins/Blur'

export default {
  name: 'AccordionModal',
  mixins: [Blur],
  props: {
    header: {
      type: String,
      default: '',
    },
    content: {
      type: String,
      default: '',
    },
    footer: {
      type: String,
      default: '',
    },
  },
  methods: {
    closeModal () {
      this.$emit('close')
      this.$parent.$emit('close')
      this.unBlur()
    },
  },
}
</script>
