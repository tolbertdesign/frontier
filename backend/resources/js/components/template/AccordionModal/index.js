import AccordionModal from './AccordionModal.vue'

export default AccordionModal
