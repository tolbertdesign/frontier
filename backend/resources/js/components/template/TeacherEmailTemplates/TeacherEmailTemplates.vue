<template>
  <AccordionModal :header="language">
    <template>
      <div class="min-h-0 max-h-full">
        <Accordion
          v-for="(email, key, index) in emails[language]"
          :key="key"
          :title="email.title"
          :is-open="index === 0"
        >
          <template slot="title">
            <p class="text-sm sm:text-base md:text-lg text-left">
              <span class="font-semibold">{{ email.title }}</span>
            </p>
          </template>

          <template>
            <div class="p-2">
              <div class="modal-card-body bg-white p-4 md:p-6 text-sm sm:text-base md:text-lg">
                <Component
                  :is="email.component"
                  :props="props"
                />
              </div>
            </div>
          </template>
        </Accordion>
      </div>
    </template>
  </AccordionModal>
</template>

<script>
import Accordion from '@/components/element/Accordion'
import AccordionModal from '@/components/template/AccordionModal'
import currency from '@/filters/currency'
import Blur from '@/mixins/Blur'
import EnglishTeamDay from '@/components/element/TeacherEmails/English/TeamDay'
import EnglishTeamDayTwo from '@/components/element/TeacherEmails/English/TeamDayTwo'
import EnglishWeekendChallenge from '@/components/element/TeacherEmails/English/WeekendChallenge'
import EnglishDayAfterFunrun from '@/components/element/TeacherEmails/English/DayAfterFunrun'
import EnglishDayBeforeFunrun from '@/components/element/TeacherEmails/English/DayBeforeFunrun'
import SpanishTeamDay from '@/components/element/TeacherEmails/Spanish/TeamDay'
import SpanishTeamDayTwo from '@/components/element/TeacherEmails/Spanish/TeamDayTwo'
import SpanishWeekendChallenge from '@/components/element/TeacherEmails/Spanish/WeekendChallenge'
import SpanishDayAfterFunrun from '@/components/element/TeacherEmails/Spanish/DayAfterFunrun'
import SpanishDayBeforeFunrun from '@/components/element/TeacherEmails/Spanish/DayBeforeFunrun'

export default {
  name: 'TeacherEmailTemplates',
  status: 'prototype',
  release: '1.0.0',
  filters: {
    currency: currency,
  },
  components: {
    Accordion,
    AccordionModal,
    EnglishTeamDay,
    EnglishTeamDayTwo,
    EnglishWeekendChallenge,
    EnglishDayAfterFunrun,
    EnglishDayBeforeFunrun,
    SpanishTeamDay,
    SpanishTeamDayTwo,
    SpanishWeekendChallenge,
    SpanishDayAfterFunrun,
    SpanishDayBeforeFunrun,
  },
  mixins: [
    Blur,
  ],
  props: {
    language: {
      type: String,
      default: 'english',
    },
    program: {
      type: Object,
      default: () => {},
    },
  },
  computed: {
    lang () {
      return this.$store.state.lang
    },
    class_pledge_total () {
      return currency(this.$store.state.User.class_pledge_total, true)
    },
    emails () {
      return {
        english: [
          {
            title: this.lang.teacher_email_templates.english.team_day_email.title,
            component: 'EnglishTeamDay',
          },
          {
            title: this.lang.teacher_email_templates.english.team_day_two_email.title,
            component: 'EnglishTeamDayTwo',
          },
          {
            title: this.lang.teacher_email_templates.english.weekend_challenge_email.title,
            component: 'EnglishWeekendChallenge',
          },
          {
            title: this.lang.teacher_email_templates.english.day_before_funrun_email.title,
            component: 'EnglishDayAfterFunrun',
          },
          {
            title: this.lang.teacher_email_templates.english.day_after_funrun_email.title,
            component: 'EnglishDayBeforeFunrun',
          },
        ],
        spanish: [
          {
            title: this.lang.teacher_email_templates.spanish.team_day_email.title,
            component: 'SpanishTeamDay',
          },
          {
            title: this.lang.teacher_email_templates.spanish.team_day_two_email.title,
            component: 'SpanishTeamDayTwo',
          },
          {
            title: this.lang.teacher_email_templates.spanish.weekend_challenge_email.title,
            component: 'SpanishWeekendChallenge',
          },
          {
            title: this.lang.teacher_email_templates.spanish.day_before_funrun_email.title,
            component: 'SpanishDayAfterFunrun',
          },
          {
            title: this.lang.teacher_email_templates.spanish.day_after_funrun_email.title,
            component: 'SpanishDayBeforeFunrun',
          },
        ],
      }
    },
    props () {
      return {
        event_name: this.program.event_name,
        registration_code: this.program.registration_code,
        due_date: this.formattedDueDate(this.program.due_date),
        link: this.program.unit.domain,
        class_pledge_total: this.class_pledge_total,
      }
    },
  },
  methods: {
    formattedDueDate (date) {
      const dueDate = new Date(date)
      return (dueDate.getMonth() + 1) + '/' + dueDate.getDate() + '/' + dueDate.getFullYear()
    },
    closeModal () {
      this.$emit('close')
      this.unBlur()
    },
  },
}
</script>
