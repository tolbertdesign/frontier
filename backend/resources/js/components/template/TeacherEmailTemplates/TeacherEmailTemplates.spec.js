import { shallowMount, createLocalVue } from '@vue/test-utils'
import TeacherEmailTemplates from './TeacherEmailTemplates.vue'
import Accordion from '@/components/element/Accordion'
import AccordionModal from '@/components/template/AccordionModal'
import EnglishTeamDay from '@/components/element/TeacherEmails/English/TeamDay'
import EnglishTeamDayTwo from '@/components/element/TeacherEmails/English/TeamDayTwo'
import EnglishWeekendChallenge from '@/components/element/TeacherEmails/English/WeekendChallenge'
import EnglishDayAfterFunrun from '@/components/element/TeacherEmails/English/DayAfterFunrun'
import EnglishDayBeforeFunrun from '@/components/element/TeacherEmails/English/DayBeforeFunrun'
import SpanishTeamDay from '@/components/element/TeacherEmails/Spanish/TeamDay'
import SpanishTeamDayTwo from '@/components/element/TeacherEmails/Spanish/TeamDayTwo'
import SpanishWeekendChallenge from '@/components/element/TeacherEmails/Spanish/WeekendChallenge'
import SpanishDayAfterFunrun from '@/components/element/TeacherEmails/Spanish/DayAfterFunrun'
import SpanishDayBeforeFunrun from '@/components/element/TeacherEmails/Spanish/DayBeforeFunrun'
import Buefy from 'buefy'
import Blur from '@/mixins/Blur'

const localVue = createLocalVue()
localVue.use(Buefy, {
  defaultIconPack: 'fas',
})
localVue.filter('currency', jest.fn())

const program = {
  id: 7,
  event_name: 'event name',
  due_date: new Date(),
  unit: {
    domain: 'test.funrun.com',
  },
  registration_code: 123456,
}

const teacher_email_templates = {
  english: {
    team_day_email: {
      title: 'Team Day Email',
    },
    team_day_two_email: {
      title: 'Team Day 2 Email',
    },
    weekend_challenge_email: {
      title: 'Weekend Challenge Email',
    },
    day_before_funrun_email: {
      title: 'Day Before Event Email',
    },
    day_after_funrun_email: {
      title: 'Day After Event Email',
    },
  },
  spanish: {
    team_day_email: {
      title: 'Spanish Team Day Email',
    },
    team_day_two_email: {
      title: 'Spanish Team Day 2 Email',
    },
    weekend_challenge_email: {
      title: 'Spanish Weekend Challenge Email',
    },
    day_before_funrun_email: {
      title: 'Spanish Day Before Event Email',
    },
    day_after_funrun_email: {
      title: 'Spanish Day After Event Email',
    },
  },
}

const $store = {
  state: {
    User: {
      programs: [
        program,
      ],
      class_pledge_total: 1200,
    },
    lang: {
      teacher_email_templates: teacher_email_templates,
    },
  },
}

describe('TeacherEmailTemplates', () => {
  it('Snapshop test for TeacherEmailTemplates for English emails', () => {
    const wrapper = shallowMount(TeacherEmailTemplates, {
      localVue,
      mixins: {
        Blur,
      },
      mocks: {
        $store,
      },
      components: {
        Accordion,
        AccordionModal,
        EnglishTeamDay,
        EnglishTeamDayTwo,
        EnglishWeekendChallenge,
        EnglishDayAfterFunrun,
        EnglishDayBeforeFunrun,
        SpanishTeamDay,
        SpanishTeamDayTwo,
        SpanishWeekendChallenge,
        SpanishDayAfterFunrun,
        SpanishDayBeforeFunrun,
      },
      propsData: {
        language: 'english',
        program: program,
      },
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('Snapshop test for TeacherEmailTemplates for Spanish emails', () => {
    const wrapper = shallowMount(TeacherEmailTemplates, {
      localVue,
      mixins: {
        Blur,
      },
      mocks: {
        $store,
      },
      components: {
        Accordion,
        AccordionModal,
        EnglishTeamDay,
        EnglishTeamDayTwo,
        EnglishWeekendChallenge,
        EnglishDayAfterFunrun,
        EnglishDayBeforeFunrun,
        SpanishTeamDay,
        SpanishTeamDayTwo,
        SpanishWeekendChallenge,
        SpanishDayAfterFunrun,
        SpanishDayBeforeFunrun,
      },
      propsData: {
        language: 'spanish',
        program: program,
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
