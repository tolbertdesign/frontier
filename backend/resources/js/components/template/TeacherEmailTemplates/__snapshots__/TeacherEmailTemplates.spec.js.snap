// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeacherEmailTemplates Snapshop test for TeacherEmailTemplates for English emails 1`] = `
<accordionmodal-stub header="english" content="" footer="">
  <div class="min-h-0 max-h-full">
    <accordion-stub title="Team Day Email" content=""><template></template>
      <div class="p-2">
        <div class="modal-card-body bg-white p-4 md:p-6 text-sm sm:text-base md:text-lg">
          <englishteamday-stub props="[object Object]"></englishteamday-stub>
        </div>
      </div>
    </accordion-stub>
    <accordion-stub title="Team Day 2 Email" content=""><template></template>
      <div class="p-2">
        <div class="modal-card-body bg-white p-4 md:p-6 text-sm sm:text-base md:text-lg">
          <englishteamdaytwo-stub props="[object Object]"></englishteamdaytwo-stub>
        </div>
      </div>
    </accordion-stub>
    <accordion-stub title="Weekend Challenge Email" content=""><template></template>
      <div class="p-2">
        <div class="modal-card-body bg-white p-4 md:p-6 text-sm sm:text-base md:text-lg">
          <englishweekendchallenge-stub props="[object Object]"></englishweekendchallenge-stub>
        </div>
      </div>
    </accordion-stub>
    <accordion-stub title="Day Before Event Email" content=""><template></template>
      <div class="p-2">
        <div class="modal-card-body bg-white p-4 md:p-6 text-sm sm:text-base md:text-lg">
          <englishdayafterfunrun-stub props="[object Object]"></englishdayafterfunrun-stub>
        </div>
      </div>
    </accordion-stub>
    <accordion-stub title="Day After Event Email" content=""><template></template>
      <div class="p-2">
        <div class="modal-card-body bg-white p-4 md:p-6 text-sm sm:text-base md:text-lg">
          <englishdaybeforefunrun-stub props="[object Object]"></englishdaybeforefunrun-stub>
        </div>
      </div>
    </accordion-stub>
  </div>
</accordionmodal-stub>
`;

exports[`TeacherEmailTemplates Snapshop test for TeacherEmailTemplates for Spanish emails 1`] = `
<accordionmodal-stub header="spanish" content="" footer="">
  <div class="min-h-0 max-h-full">
    <accordion-stub title="Spanish Team Day Email" content=""><template></template>
      <div class="p-2">
        <div class="modal-card-body bg-white p-4 md:p-6 text-sm sm:text-base md:text-lg">
          <spanishteamday-stub props="[object Object]"></spanishteamday-stub>
        </div>
      </div>
    </accordion-stub>
    <accordion-stub title="Spanish Team Day 2 Email" content=""><template></template>
      <div class="p-2">
        <div class="modal-card-body bg-white p-4 md:p-6 text-sm sm:text-base md:text-lg">
          <spanishteamdaytwo-stub props="[object Object]"></spanishteamdaytwo-stub>
        </div>
      </div>
    </accordion-stub>
    <accordion-stub title="Spanish Weekend Challenge Email" content=""><template></template>
      <div class="p-2">
        <div class="modal-card-body bg-white p-4 md:p-6 text-sm sm:text-base md:text-lg">
          <spanishweekendchallenge-stub props="[object Object]"></spanishweekendchallenge-stub>
        </div>
      </div>
    </accordion-stub>
    <accordion-stub title="Spanish Day Before Event Email" content=""><template></template>
      <div class="p-2">
        <div class="modal-card-body bg-white p-4 md:p-6 text-sm sm:text-base md:text-lg">
          <spanishdayafterfunrun-stub props="[object Object]"></spanishdayafterfunrun-stub>
        </div>
      </div>
    </accordion-stub>
    <accordion-stub title="Spanish Day After Event Email" content=""><template></template>
      <div class="p-2">
        <div class="modal-card-body bg-white p-4 md:p-6 text-sm sm:text-base md:text-lg">
          <spanishdaybeforefunrun-stub props="[object Object]"></spanishdaybeforefunrun-stub>
        </div>
      </div>
    </accordion-stub>
  </div>
</accordionmodal-stub>
`;
